src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -172,8 +182,23 @@ public byte[] downloadAndReadOneUrl(
   }
 
   private HttpConnectorMultiplexer setUpConnectorMultiplexer(
-      ExtendedEventHandler eventHandler, Map<String, String> clientEnv) {
+      ExtendedEventHandler eventHandler, Map<String, String> clientEnv) throws IOException {
     ProxyHelper proxyHelper = new ProxyHelper(clientEnv);
+
+    @Nullable SSLContext maybeSslContext;
+    try {
+      maybeSslContext = SSLContextBuilder.build(authAndTLSOptions);
+    } catch (IOException e) {
+      // This should be a hard error.  However, the HTTP downloader did NOT try to use the TLS
+      // credentials when it was first written and the attempt to create an SSL context could fail
+      // while parsing the configured certificates (if any).  By ignoring the error, we don't
+      // regress behavior: builds that would fail due to problems with downloads before would still
+      // fail in the same way.
+      logger.atWarning().withCause(e)
+          .log("Not using configured TLS credentials in the HTTP downloader");
+      maybeSslContext = null;
```

Comment:
Does this SSLContext completely override the default certificate store, or does it add to it? (Unfortunately, I'm not super familiar with how SSL certificates work in Java.)

My concern is that we're reusing command line flags that some folks are already using to set up a custom TLS certificate for remote caching/execution. If that custom certificate isn't generally appropriate for HTTPS connections, is there a risk that we could break repository downloads for these users?

Reply:
I don't _think_ it does. Our build is full of https URLs that don't require a certificate but require certificate validation, like pretty much every other build out there. If this did override the defaults, I'd imagine those downloads would start breaking, but they don't...

Looking at SSLContextBuilder, we are calling `init` on the `SSLContext` with the default trust store but with a custom key store. So this seems to validate the previous: we now provide some keys (the default doesn't) but we rely on the same trust chains as before.

Reply:
Resolving because I think the original question has been answered.